activerecord/lib/active_record/attribute_methods/query.rb

```diff
@@ -16,9 +16,8 @@ def query_attribute(attr_name)
         when true        then true
         when false, nil  then false
         else
-          column = self.class.columns_hash[attr_name]
-          if column.nil?
-            if Numeric === value || value !~ /[^0-9]/
+          if !type_for_attribute(attr_name) { false }
+            if Numeric === value || (String === value && value !~ /[^0-9]/)
```

Comment:
Changing this line is not covered any test, in other words, all new test cases will be fixed by the `if !type_for_attribute(attr_name) { false }` line.

Reply:
I will fix it and below comment points also. 🙇